### ISSUE TITLE
dom class value lost in a special case

### ISSUE THREAD

**Issue by zhenkunou** [developit/preact#181]
hi~ @developit 

in this codepen:
http://codepen.io/ryan_ou/pen/rLOaJa?editors=0010 (5.0.0-beta9)

follow these steps you will see the issue:
1. click Set background button
2. click Hide/Show button 4 times
3. in chrome dev tools - Elements tab you will see:
```html
<div id="panel" class>
  <button>SetBackground true</button>
</div>
```
it should be:
```html
<div id="panel" class='bg'>
  <button>SetBackground true</button>
</div>
```

If you change the vnode(id='panel') to be:
```html
<div id='panel' className={hasBg ? 'bg' : 'a_not_exists_class'}>...</div>
```
the class will be correct.

It seems like that:
If a `component.className` has been set to `null/undefined/''`;
a dom's class will lost when preact re-use the `component.base`

**Comment by developit** [comment]
Good catch. I believe this line is the suspect: 
https://github.com/developit/preact/blob/master/src/vdom/diff.js#L248

**Comment by developit** [comment]
Ahh - @zhenkunou looks like this is specific to `className` - the bug doesn't seem to occur when using `class`.

**Comment by developit** [comment]
Fix released in `5.0.0-beta10`.

**Comment by zhenkunou** [comment]
I've update the demo(5.0.0-beta10), it work now!! Great, thx you
